### ISSUE TITLE
Show CGM life left on main screen

### ISSUE THREAD

**Issue by JohnKitching** [nightscout/Trio#304]
In the same way that life left of a pump is displayed in the main screen it would be very handy to see the same for the CGM. 

I have no opinion on how it's displayed, just that I'd like to see it. 

It would help with planning CGM changes around a busy life. I'm guessing this would help most people.

**Comment by LiroyvH** [comment]
Not sure how much life the current design has left in it, its a bit short on space. But maybe it'd be doable to implement a countdown bar thingy like xDrip4iOS has. You'd sacrifice a bit of graph space, but I think we have plenty of that. Alternatively the BG-label in the legend could be dropped and the eventual glucose moved to top right (left: basalrate, right: eventual glucose) to free up some space at the bottom. Get's crowded tho.

**Comment by dnzxy** [comment]
Like the idea a lot. How do you guys imagine to handle Anubis and extended sensors in this type of display? 🤔

**Comment by JohnKitching** [comment]
I'd say that if it's possible, if there's a known life then display it, if there isn't (Anubis etc) then display nothing.

**Comment by bjornoleh** [comment]
Here are some samples of how xDrip4iOS handles Anubis (G6) before and ten days. 

Before:
<img src="https://github.com/nightscout/Trio/assets/63544115/604e317c-e2be-4c22-b870-9f73d4db296e" width="300">


After ten days:
<img src="https://github.com/nightscout/Trio/assets/63544115/0b7d95ac-57ce-4368-b200-7e765ec42203" width="300">

**Comment by LiroyvH** [comment]
I'd suggest that for Anubis either we show nothing or indeed just keep the counter going after 10 days. Anubis can be detected easily by the 180 days in the txAge, stock Dexcom won't ever have that. For my Anubis I'm typically more interested in the VoltB than SAGE. :P

**Comment by bjornoleh** [comment]
This could still be an interesting feature. I set the `long-term` label now.

**Comment by dnzxy** [comment]
Maybe this would be something down your lane and you'd want to have a look at this, once public beta drops @aidanlane 😎

**Comment by aidanlane** [comment]
Thanks for thinking of me @dnzcy, but I now have another feature that I’d like to focus on first 😉

**Comment by LiroyvH** [comment]
@Sjoerd-Bo3 That's horrible for OCD with the alignment and mismatching amount of rows haha :) Might it be possible to move the adjustments info section up a tiny little bit and then below that put a countup or countdown bar for sensor age? (Like how xDrip4iOS has it implemented).

Otherwise I'd suggest moving the predicted eventual glucose value to the area where currently the holy trinity is positioned (IOB, COB, current basal) and use the freed up space for the sensor age the way you've put it on the left now. 
That way you keep the alignment in place, you keep the same amount of items on each side of the glucose value and it can be argued that eventual glucose is not out of place in that row either.

**Comment by Sjoerd-Bo3** [comment]
> [@Sjoerd-Bo3](https://github.com/Sjoerd-Bo3) That's horrible for OCD with the alignment and mismatching amount of rows haha :) Might it be possible to move the adjustments info section up a tiny little bit and then below that put a countup or countdown bar for sensor age? (Like how xDrip4iOS has it implemented).
> 
> Otherwise I'd suggest moving the predicted eventual glucose value to the area where currently the holy trinity is positioned (IOB, COB, current basal) and use the freed up space for the sensor age the way you've put it on the left now. That way you keep the alignment in place, you keep the same amount of items on each side of the glucose value and it can be argued that eventual glucose is not out of place in that row either.

I agree 100% on the OCD part, that's why I'm asking :)

**Comment by JohnKitching** [comment]
That looks fine to me thank you. 
I suppose you could argue it should be inside the ring with the other CGM stuff. 
But just being there is what's important to me.

**Comment by Sjoerd-Bo3** [comment]
@LiroyvH  But that doesn't work for smaller screens. And I think it is best to keep Pump/CGM info together and not also at the bottom of the view.

Also the 'holy trinity' was reduced from 4 items to 3 because of space constraints of smaller screens. But I agree on the notion of moving eventual glucose to somewhere else.

**Comment by Sjoerd-Bo3** [comment]
> That looks fine to me thank you. I suppose you could argue it should be inside the ring with the other CGM stuff. But just being there is what's important to me.

Also true, but how would that work xD

**Comment by JohnKitching** [comment]
Personally, I'd prefer remaining time but only if it's accurate. With L2 it looks like the end time can be read from the interface but I have no idea for anything else.

**Comment by bjornoleh** [comment]
> What do we prefer? elapsed or remaining? Remaining right? like the pump.
> 
> But we cannot fit a bar there, so how to make it work....

A one or two pixle wide bar like this might fit? Not sure how many pixles this is, but just as an illustration. I think something like this would pretty celarly read as elapsed time (the colour at the left hand, grayish on the right). Would neeed to be mirrored for right to left hand reading languages of course.

<img width="650" height="1400" alt="Image" src="https://github.com/user-attachments/assets/8d760337-4c8d-42f2-a93d-46c026e307b4" />

**Comment by Sjoerd-Bo3** [comment]
And then also for the pump? to balance it?

**Comment by Sjoerd-Bo3** [comment]
@dnzxy I agree, already mentioned it being 4 before was not preferred. But whats is the solution then?

**Comment by Sjoerd-Bo3** [comment]
> You‘d need to somewhat incorporate the CGM lifetime into the center CGM-related UI element, which is the "glucose bobble". No immediate pointer to offer for it, sorry.
> 
> I can just say that the proposal in the comments here is really "breaking" a bunch of concepts, ideas and many iterations of back and forth with testers and UI/UX folks we went through, which is why I semi spoke out against it.

I agree with this, I’m gonna search for a good implementation in the glucose bobble. Will get back to that. For this feature to work I need to make some extra additions to xdrip aswell. So will need some work anyway.

**Comment by JohnKitching** [comment]
How about a countdown circle inside the outer ring?

**Comment by LiroyvH** [comment]
I honestly can't see this working in the glucose circle; though i'll be curious what you come up with. Its already rather crammed (and I say that from a 16 Pro) as it is. 

Incidentally, when you tap the glucose: it will open the CGM information. This immediately reveals the sensor age. Isn't that sufficient? Does it *have* to be on the screen constantly? If it really must be, another venue to work-around the real-estate issue is by alternating the pod lifetime (or on MDT/DANA: battery level) and CGM lifetime. Eg: every few seconds, the information changes from battery/pod lifetime to CGM lifetime and a few seconds later it goes back again?

**Comment by antoinekh** [comment]
It is easier to know when the sensor will expire (and not be caught off guard) when the information is directly visible.
Perhaps put the value under loop status as @bjornoleh propose and move the  glucose target estimation (-> with glucose value) inside the glucose bubble

**Comment by polyfeux** [comment]
I'm currently on dev and my Dexcom G7 is expiring today. 
So I get a firsthand experience on the colored arc displaying the CGM lifetime. 

The color kind of confuses me, because the red number says 3h (and later 2h), while the red arc is getting bigger - not smaller (as the time does). 
Should the coloring not be the other way - so the gray and the red areas of the arc swapped?

/edit: the text with the remaining hours is a bit too close to the COB above the diagram. 

~And does the arc with the remaining time needs to be really displayed at all times (displaying the bright green clashes with color-concept of Trio IMHO) or only as soon as the 24/48h period starts?~

<img width="1169" height="619" alt="Image" src="https://github.com/user-attachments/assets/568e3eea-09cc-4265-aeaa-91ae9ace7497" />

**Comment by dnzxy** [comment]
@polyfeux Text size is being adjusted in a much bigger rework of home screen that’s in the works. 

We are orienting ourselves here with how iOS typically shows things progressing, here lifetime bearing its end. It could be inversed, but here the full circle is to be considered a full sensor lifetime that‘s reached. 

The arc is shown only at 48 hrs of lifetime, and during grace period.

**Comment by dnzxy** [comment]
@polyfeux have a look at #1331 , addressed your suggestion there. 

Side note: I will be closing this issue, as this has been addressed with #1205
